src/Oceanostics.jl

```diff
@@ -72,8 +72,10 @@ end
 #---
 
 #+++ Utils for removing mean fields
-using Oceananigans: prognostic_fields
+using Oceananigans: prognostic_fields, HydrostaticFreeSurfaceModel
 using Oceananigans.Biogeochemistry: biogeochemical_auxiliary_fields
+import Base: -
+-(a::ZeroField, b::ZeroField) = a
```

Comment:
Add to Oceananigans?

Reply:
Yeah, if you anticipate that could be useful outside Oceanostics.

Reply:
It's also a generic principle not to extend methods for types in a package that doesn't "own" those types (it's called type piracy, because you are "pirating" the behavior of the types 😄 ). Basically people might reimplement or (for some reason) implement different behavior in Oceananigans sometime in the future. Then this package would conflict with Oceananigans and could become unuseable... so for the benefit of all, best to extend methods inside the package that owns the types involved.

Obviously a pretty minor situation here, but its a teaching moment

Reply:
https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-type-piracy

Reply:
Lesson learned! I can add a small PR to Oceananigans, or avoid it by trimming the `mean_fields` beforehand:
```julia
mean_fields = NamedTuple(key => val for (key,val) in zip(keys(mean_fields), mean_fields) if !(val isa ZeroField))
```
Which is better?

Reply:
Simpler is

```julia
mean_fields = NamedTuple(key => val for (key, val) in pairs(mean_fields) if !(val isa ZeroField))
```

But

```julia
-(a::ZeroField, b::ZeroField) = a
```

seems simpler enough a little neater. Only question I have, do you need to worry about the types of `a` and `b`? Cause `ZeroField` has an eltype:

https://github.com/CliMA/Oceananigans.jl/blob/1ee317b9b6986436f18dd12d7fcdd7067d9234b7/src/Fields/constant_field.jl#L5

seems like to do define the arithmetic well you need to promote types or something

Reply:
```julia
function Base.-(z1::ZeroField{T1, N1}, z2::ZeroField{T2, N2}) where {T1, T2, N1, N2}
    T = Base.promote_type(T1, T2)
    N = max(N1, N2)
    return ZeroField{T, N}()
end
```

and heck while you're at it why not do a bit more

```julia
import Base: -, +, *

for op in (-, +, *)
    @eval begin
        function $(op)(z1::ZeroField{T1, N1}, z2::ZeroField{T2, N2}) where {T1, T2, N1, N2}
            T = Base.promote_type(T1, T2)
            N = max(N1, N2)
            return ZeroField{T, N}()
        end
    end
end
```

I think that's the right syntax but not 100% sure

Reply:
`ZeroField` is implemented here: https://github.com/CliMA/Oceananigans.jl/blob/main/src/Fields/constant_field.jl#L5

Reply:
Thanks! I will give it a try.

Reply:
Just for completeness, this is being implemented in https://github.com/CliMA/Oceananigans.jl/pull/3567